### PR TITLE
Add guideline for namespace documentation

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds a guideline on what to do regarding namespaces in IDL.

### CODE REVIEW

**Review by Elchi3 — APPROVED** [review]
Thanks @queengooborg. LGTM!

cc'ing another owner to approve the guideline change. @foolip, do you agree?

**Comment by Elchi3** [comment]
@foolip ping

File: docs/data-guidelines/api.md [hunk 1]
```diff
@@ -26,6 +26,10 @@ Members of this mixin are available to `HTMLAnchorElement` and `HTMLAreaElement`
 
 This guideline was proposed in [#8929](https://github.com/mdn/browser-compat-data/issues/8929), based in part on previous discussion in [#472](https://github.com/mdn/browser-compat-data/issues/472).
 
+## Namespaces
+
+[Namespaces](https://webidl.spec.whatwg.org/#idl-namespaces) in Web IDL are similar to interfaces. A namespace should only be documented in BCD if it contains properties or methods that would normally be documented. For example, the `console` namespace contains several properties and methods, so it should be documented in BCD. The `GPUBufferUsage` namespace only contains constants, which should not be documented in BCD; thus, the `GPUBufferUsage` namespace should not be documented.
```
Comment (foolip) [inline docs/data-guidelines/api.md:31]:
```suggestion
[Namespaces](https://webidl.spec.whatwg.org/#idl-namespaces) in Web IDL are similar to interfaces. A namespace should only be documented in BCD if it contains properties or functions that would normally be documented. For example, the `console` namespace contains many functions, so it should be documented in BCD. The `GPUBufferUsage` namespace only contains constants, which should not be documented in BCD; thus, the `GPUBufferUsage` namespace should not be documented.
```

**Review by foolip — APPROVED** [review]
Some wording suggestions, but I agree with the guideline!

Note that namespaces don't have methods in the same way that interfaces do, they're effectively static functions, and the `this` value doesn't matter.

Also there are no attributes in https://console.spec.whatwg.org/

**Pull request merged**